tests/misc-test.cc

```diff
@@ -216,13 +216,56 @@ void check_capability_set_inexact_at_most()
 	free(p);
 }
 
+void check_sealed_scoping()
+{
+	Capability<void> o{switcher_current_thread()};
+	TEST(o.is_valid() && (o.type() == CheriSealTypeSealedTrustedStacks),
+	     "Shared object cap not as expected: {}",
+	     o);
+
+	// Take the address of the o cap, requiring that it go out to memory.
+	Capability<Capability<void>> oP{&o};
+
+	/*
+	 * Load a copy of our sealed o cap through an authority that lacks
+	 * LoadGlobal permission.  The result should be identical to the original
+	 * but without global permission.
+	 */
+	Capability<Capability<void>> oPNoLoadGlobal = oP;
+	oPNoLoadGlobal.permissions() &=
+	  PermissionSet::omnipotent().without(Permission::LoadGlobal);
+	Capability<void> oLocal1 = *oPNoLoadGlobal;
+
+	TEST(
+	  oLocal1.is_valid() && (oLocal1.type() == o.type()) &&
+	    (oLocal1.permissions() == o.permissions().without(Permission::Global)),
+	  "Loading global sealed cap through non-LoadGlobal authority gone wrong: "
+	  "{} {}",
+	  o,
+	  oLocal1);
+
+	/*
+	 * Use CAndPerm to shed Global from our o cap.
+	 * Spell this a little oddly to make sure we get CAndPerm with a mask of
+	 * all 1s but Global.  Using oLocal2.permissions().without() would do a
+	 * cgetperm and then candperm.
+	 */
+	Capability<void> oLocal2 = o;
+	oLocal2.permissions() &=
+	  PermissionSet::from_raw(-1).without(Permission::Global);
```

Comment:
I'm a little worried that even this spelling isn't quite enough.  I'd not used `::omnipotent()` because I failed to notice that `::from_raw()` _also_ masks with `valid_permission_mask()` and so I'm worried that we're presently relying on the compiler to know that bits above those set by `valid_permission_mask()` are irrelevant?  Maybe we should have a `PermissionSet::allButGlobal` constant that works for the stringent requirement of https://github.com/CHERIoT-Platform/cheriot-sail/pull/83 ?